src/main/java/cnLabs/fileStorageAPI/Services/FileService.java

```diff
@@ -34,7 +34,7 @@ public DatabaseFile getFileByName(String name) {
     }
 
     public URI saveFile(MultipartFile file) throws IOException {  // How will this IOException get handled?
```

Comment:
The Exception would be propagated up the call stack until it is `catch`ed. 
In case of this app it would be `FileController` the `uploadFile` method, line 53:
```java
    @PostMapping
    public ResponseEntity<?> uploadFile(@RequestBody MultipartFile file) {
        try {
            return ResponseEntity.created(fileService.saveFile(file)).build();
        } catch (Exception e) { // <- here
            return ResponseEntity.badRequest().body(e.getMessage());
        }
    }
```
The `400 Bad Request` response would be returned. 

But more generally, let's say you wouldn't have the `catch` like here:
```java
    @PostMapping
    public ResponseEntity<?> uploadFile(@RequestBody MultipartFile file) throws IOException {
        return ResponseEntity.created(fileService.saveFile(file)).build();
    }
```
Spring has a global `catch` wrapping all controllers (I bet you've already received a couple `500`, right). This allows the Spring server to continue functioning even if one of the controllers/services/repositories/etc., fails with an error. 

There is also a way to take leverage of the global `catch` I'll show you during 1-1 :)

Reply:
Thanks! Makes sense.